src/libcore/iter/adapters/fuse.rs

```diff
@@ -201,7 +172,131 @@ macro_rules! unchecked {
 }
 
 #[stable(feature = "fused", since = "1.26.0")]
-impl<I> Iterator for Fuse<I>
+impl<I> Iterator for Fuse<I> where I: FusedIterator {}
+
+#[stable(feature = "fused", since = "1.26.0")]
+impl<I> DoubleEndedIterator for Fuse<I> where I: DoubleEndedIterator + FusedIterator {}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<I> ExactSizeIterator for Fuse<I> where I: ExactSizeIterator + FusedIterator {}
```

Comment:
Do we still need these impls? I think not, and I'm surprised this is accepted at all, but I guess they're acting as a sort of empty specialization here. The `stable` aspect should still be fine, since they're covered by the main impl.

Reply:
Removed the empty impls